### DISCUSSION TITLE
Any plans for adding support for sparse files?

### DISCUSSION THREAD

**Discussion by tastewar** [ned14/llfio#142] [Q&A] [answered]
I am looking forward to using llfio to port a project that was originally written for Windows and which uses unbuffered i/o to be able to run on Windows and Linux. One thing that would make that job easier would be to be able to mark blocks as sparse and query the file about sparse ranges within the file. Just curious if you have plans to add any functions for that. Thanks!

**Comment by ned14** [comment, marked as answer]
Like `file_handle::zero()` and `file_handle::extents()`?

**Reply by tastewar** [reply]
Oh! Yes, a *lot* like that! :-)

All my searching was on sparse, and I was coming up empty. Thank you!

**Reply by tastewar** [reply]
(I can't figure out how to mark this as the answer...)

**Reply by BurningEnlightenment** [reply]
You can only mark answers as accepted in the Q&A section
